src/main/drivers/serial_uart_stm32g4xx.c

```diff
@@ -0,0 +1,444 @@
+        .txBufferSize = sizeof(uart3TxBuffer),
+        .rxBufferSize = sizeof(uart3RxBuffer),
+    },
+#endif
+
+#ifdef USE_UART4
+    {
+        .device = UARTDEV_4,
+        .reg = UART4,
+#ifdef USE_DMA
+        .rxDMAChannel = DMA_REQUEST_UART4_RX,
+        .rxDMAResource = (dmaResource_t *)UART4_RX_DMA_CHANNEL,
+        .txDMAChannel = DMA_REQUEST_UART4_TX,
+        .txDMAResource = (dmaResource_t *)UART4_TX_DMA_CHANNEL,
+#endif
+        .rxPins = {
+            { DEFIO_TAG_E(PC11), GPIO_AF5_UART4 },         
+        },
+        .txPins = {
+            { DEFIO_TAG_E(PC10), GPIO_AF5_UART4 },         
+        },
+        .rcc_apb1 = RCC_APB11(UART4),
+        .rxIrq = UART4_IRQn,
+        .txPriority = NVIC_PRIO_SERIALUART4_TXDMA,
+        .rxPriority = NVIC_PRIO_SERIALUART4,
+        .txBuffer = uart4TxBuffer,
+        .rxBuffer = uart4RxBuffer,
+        .txBufferSize = sizeof(uart4TxBuffer),
+        .rxBufferSize = sizeof(uart4RxBuffer),
+    },
+#endif
+
+#ifdef USE_UART5
+    {
+        .device = UARTDEV_5,
+        .reg = UART5,
+#ifdef USE_DMA
+        .rxDMAChannel = DMA_REQUEST_UART5_RX,
+        .rxDMAResource = (dmaResource_t *)UART5_RX_DMA_CHANNEL,
+        .txDMAChannel = DMA_REQUEST_UART5_TX,
+        .txDMAResource = (dmaResource_t *)UART5_TX_DMA_CHANNEL,
+#endif
+        .rxPins = {
+            { DEFIO_TAG_E(PD2),  GPIO_AF5_UART5 },
+        },
+        .txPins = {
+            { DEFIO_TAG_E(PC12), GPIO_AF5_UART5 },
+        },
+        .rcc_apb1 = RCC_APB11(UART5),
+        .rxIrq = UART5_IRQn,
+        .txPriority = NVIC_PRIO_SERIALUART5_TXDMA,
+        .rxPriority = NVIC_PRIO_SERIALUART5,
+        .txBuffer = uart5TxBuffer,
+        .rxBuffer = uart5RxBuffer,
+        .txBufferSize = sizeof(uart5TxBuffer),
+        .rxBufferSize = sizeof(uart5RxBuffer),
+    },
+#endif
+
+#ifdef USE_UART6
+    // On G474, UART6 is implemented as LPUART1
+    {
+        .device = UARTDEV_6,
+        .reg = LPUART1,
+#ifdef USE_DMA
+        .rxDMAChannel = DMA_REQUEST_LPUART1_RX,
+        .rxDMAResource = (dmaResource_t *)UART6_RX_DMA_CHANNEL,
+        .txDMAChannel = DMA_REQUEST_LPUART1_TX,
+        .txDMAResource = (dmaResource_t *)UART6_TX_DMA_CHANNEL,
+#endif
+        .rxPins = {
+            { DEFIO_TAG_E(PA3),  GPIO_AF12_LPUART1 },
+            { DEFIO_TAG_E(PB10), GPIO_AF8_LPUART1 },
+            { DEFIO_TAG_E(PC0),  GPIO_AF8_LPUART1 },
+        },
+        .txPins = {
+            { DEFIO_TAG_E(PA2),  GPIO_AF12_LPUART1 },
+            { DEFIO_TAG_E(PB11), GPIO_AF8_LPUART1 },
+            { DEFIO_TAG_E(PC1),  GPIO_AF8_LPUART1 },
+        },
+        .rcc_apb1 = RCC_APB12(LPUART1),
+        .rxIrq = LPUART1_IRQn,
+        .txPriority = NVIC_PRIO_SERIALUART6_TXDMA,
+        .rxPriority = NVIC_PRIO_SERIALUART6,
+        .txBuffer = uart6TxBuffer,
+        .rxBuffer = uart6RxBuffer,
+        .txBufferSize = sizeof(uart6TxBuffer),
+        .rxBufferSize = sizeof(uart6RxBuffer),
+    },
+#endif
+};
+
+#if 0
```

Comment:
If this code is not used then it should be removed - leaving it in commented out will just lead to rot.